ras_party/views/respondent_view.py

```diff
@@ -31,18 +32,56 @@ def get_pw(username):
 
 @respondent_view.route('/respondents', methods=['GET'])
 def get_respondents():
+    """Get respondents by id or a combination of firstName, lastName and EmailAddress
+    Note, result set for names and email includes total record count to support pagination, get by id does not
+    """
+
     ids = request.args.getlist("id")
+    first_name = request.args.get("firstName", default="").strip()
+    last_name = request.args.get("lastName", default="").strip()
+    email = request.args.get("emailAddress", default="").strip()
+    page = int(request.args.get("page", default=1))
+    limit = int(request.args.get("limit", default=10))
+
+    _validate_get_respondent_params(ids, first_name, last_name, email)
+
     if ids:
         # with_db_session function wrapper automatically injects the session parameter
         # pylint: disable=no-value-for-parameter
         response = respondent_controller.get_respondent_by_ids(ids)
     else:
-        logger.debug("The parameter id is required.", url=request.url)
-        raise BadRequest("The parameter id is required.")
-
+        # with_db_session function wrapper automatically injects the session parameter
+        # pylint: disable=no-value-for-parameter
+        response = respondent_controller.get_respondents_by_name_and_email(first_name, last_name, email, page, limit)
     return jsonify(response)
 
 
+def _validate_get_respondent_params(ids, first_name, last_name, email):
+    """
+    Validates the combination of parameters for get respondents
+    :param ids:
+    :param first_name:
+    :param last_name:
+    :param email:
+    """
+
+    if not (ids or first_name or last_name or email):
+        logger.debug("Invalid params: either id , first_name or last_name or email are required")
+        raise BadRequest(f"id , first_name or last_name or email are required")
```

Comment:
Super minor, but should be `f"id, first_name......`

Reply:
Also why is this an f string?

Reply:
Good catch, changing